### ISSUE TITLE
Add Loss functions to PyTorch Frontend

### ISSUE THREAD

**Issue by djl11** [unifyai/ivy#26]
Add [Loss functions](https://pytorch.org/docs/stable/nn.functional.html#loss-functions) to PyTorch frontend:

- [ ] binary_cross_entropy
- [ ] binary_cross_entropy_with_logits
- [ ] poisson_nll_loss
- [ ] cosine_embedding_loss
- [ ] cross_entropy
- [ ] ctc_loss
- [ ] gaussian_nll_loss
- [ ] hinge_embedding_loss
- [ ] kl_div
- [ ] l1_loss
- [ ] mse_loss
- [ ] margin_ranking_loss
- [ ] multilabel_margin_loss
- [ ] multilabel_soft_margin_loss
- [ ] multi_margin_loss
- [ ] nll_loss
- [ ] huber_loss
- [ ] smooth_l1_loss
- [ ] soft_margin_loss
- [ ] triplet_margin_loss
- [ ] triplet_margin_with_distance_loss

**Comment by VikramxD** [comment]
- [ ]  #2577

**Comment by ahlag** [comment]
https://github.com/unifyai/ivy/issues/2660

**Comment by ahmedelbehery99** [comment]
[#2679]

**Comment by hirwa-nshuti** [comment]
cross_entropy #2751

**Comment by raotnameh** [comment]
https://github.com/unifyai/ivy/issues/2902#issue-1335928169

**Comment by hirwa-nshuti** [comment]
binary_cross_entropy #3028

**Comment by whitepurple** [comment]
#3037

**Comment by WOWspring** [comment]
- [ ] #3531

**Comment by WOWspring** [comment]
- [ ] #3532

**Comment by asadimtiazmalik** [comment]
- [ ] #4180

**Comment by bicycleman15** [comment]
- [ ] https://github.com/unifyai/ivy/issues/4603

**Comment by bicycleman15** [comment]
- [ ] https://github.com/unifyai/ivy/issues/4604

**Comment by bicycleman15** [comment]
-[ ] https://github.com/unifyai/ivy/issues/4605

**Comment by bicycleman15** [comment]
- [ ] https://github.com/unifyai/ivy/issues/4606

**Comment by mibo8** [comment]
- [ ] #5083

**Comment by aditya-hari** [comment]
- [] #5302

**Comment by YueBro** [comment]
#5546

**Comment by Tatiats7** [comment]
#5741

**Comment by jfrausto7** [comment]
- mse_loss #6443

**Comment by bipinKrishnan** [comment]
- [ ] #6993

**Comment by ZiadAmerr** [comment]
#7065

**Comment by Gabcddrc** [comment]
nll_loss #7140

**Comment by satani99** [comment]
[]#7199

**Comment by satani99** [comment]
- [ ] soft_margin_loss

**Comment by AnnaTz** [comment]
- [ ] #7592

**Comment by ZiadAmerr** [comment]
- #7673

**Comment by m1rakram** [comment]
gaussian_nll_loss #7700

**Comment by satani99** [comment]
- [ ] multi_label_loss

**Comment by zaRizk7** [comment]
kl_div #9183

**Comment by ju2ez** [comment]
https://github.com/unifyai/ivy/issues/9499

**Comment by Alexeybary** [comment]
- [ ] #10198

**Comment by RaduGalan1** [comment]
- [ ] #10237

**Comment by JSSan97** [comment]
- [ ] #10295

**Comment by zysymu** [comment]
- [ ] #10383

**Comment by tomatillos** [comment]
- [ ]  #10484

**Comment by soma2000-lang** [comment]
multilabel_margin_loss #10582

**Comment by Dhruv2012** [comment]
triplet_margin_loss #10609

**Comment by yopknopixx** [comment]
Taking up `multilabel_soft_margin_loss`. #10625

**Comment by iamharsha1999** [comment]
gaussian_nll_loss #10920

**Comment by ZiadAmerr** [comment]
- [] #10930

**Comment by iamharsha1999** [comment]
#12017

**Comment by Dhruv2012** [comment]
- [ ] #12176

**Comment by Cemilcelik11** [comment]
#12745

**Comment by dorltcheng** [comment]
#14024

**Comment by guitld** [comment]
#14547

**Comment by Br-Al** [comment]
multi_margin_loss #15526

**Comment by bbyiringiro2** [comment]
- [ ] #17721

**Comment by bbyiringiro2** [comment]
- [] #17733

**Comment by bbyiringiro2** [comment]
- [ ] #17733

**Comment by Huvinesh-Rajendran-12** [comment]
- [ ] #19764

**Comment by Huvinesh-Rajendran-12** [comment]
- [ ] #20121

**Comment by nhewitt99** [comment]
- [ ] #22864

**Comment by nhewitt99** [comment]
Hey, looks like the hinge_embedding_loss was already implemented in this PR #11124 but didn't get updated in the todo tracker.

**Comment by CanKeles5** [comment]
[] #23507

**Comment by CanKeles5** [comment]
It seems like the multilabel_soft_margin_loss is already implemented, but the task is still in the ToDo list as undone.

**Comment by HemanthSai7** [comment]
Are there still any todo loss functions left?

**Comment by tbeetech** [comment]
- [ ] #24636

**Comment by Rhythm1821** [comment]
#26978

**Comment by Rhythm1821** [comment]
#26981
